src/handlers/int-test-support/helpers/mock-data/csv/objectsToCsv.ts

```diff
@@ -0,0 +1,71 @@
+interface ObjectsToCSVOptions {
+  filterKeys?: string[]; // a list of keys to be filtered out of the output csv
+  renameKeys?: Map<string, string>; // a dictionary of keys to be renamed
+}
+type ObjectsToCSV = (
+  objects: Array<Record<string, string | number>>,
+  options?: ObjectsToCSVOptions
+) => string;
+
+export const objectsToCSV: ObjectsToCSV = (objects, options) => {
+  // Filter objects with given keys
+  if (options?.filterKeys !== undefined) {
+    filterKeys(objects, options?.filterKeys);
+  }
+
+  // Set up header row
+  const csvString = [];
+  const headerRow = [
+    ...new Set(objects.map((object) => Object.keys(object)).flat()),
+  ];
+  console.log(headerRow);
+
+  // Rename header row items with given dictionary
+  if (options?.renameKeys !== undefined) {
+    csvString.push(renameHeaderKeys(headerRow, options?.renameKeys));
+  } else {
+    csvString.push(headerRow);
+  }
+
+  // Build CSV String
+  for (let i = 0; i < objects.length; i++) {
+    const row = [];
+    for (let j = 0; j < headerRow.length; j++) {
+      row.push(objects[i][headerRow[j]]);
+    }
+    csvString.push(row);
+  }
+  return csvString.map((e) => e.join(",")).join("\n");
+};
+
+function filterKeys(
+  objects: Array<Record<string, string | number>>,
+  keysToFilter: string[]
+): void {
+  keysToFilter.forEach((keyToFilter) =>
+    objects.map((object) => {
+      for (const key in object) {
+        if (key.match(keyToFilter) != null) {
+          // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
+          delete object[key];
+        }
+      }
+      return object;
+    })
+  );
+}
```

Comment:
function filterKeys(
  objects: Array<Record<string, string | number>>,
  keysToFilter: string[]
): void {
```suggestion
export const filterKeys = (
  objects: Array<Record<string, string | number>>,
  keysToFilter: string[]
): Array<Record<string, string | number>> => {
  return objects.reduce<Array<Record<string, string | number>>>(
    (filteredObjects, currentObject) => [
      ...filteredObjects,
      {
        ...Object.entries(currentObject).reduce(
          (filteredObject, [key, value]) => {
            if (keysToFilter.includes(key)) return filteredObject;
            return {
              ...filteredObject,
              [key]: value,
            };
          },
          {}
        ),
      },
    ],
    []
  );
};
```

This way you don't mutate the original object, bit more functional

Reply:
Will prob take me some time to understand how this .reduce is working here but I'll use your suggestion to avoid mutating the original object :)